### PR TITLE
Work view

### PR DESCRIPTION
**Pull request by Kwolik**
1. Create one page for work
2. Create components: MyWorkComponent, WorkForReviewComponent, CurrentVersion, CurrentVersionWithReplyToReview
3. Include a different view for Reviewer and Participant
4. Create drop-down list
5. Create popup for adding reviews
6. The drop-down list is different for Reviewer and Participant

Participant:
image
image

===================================================================================
Reviewer:
image
image

### CODE REVIEW

File: web-client/src/pages/work-view.js [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+    ]
+
+    // by default every version should be closed (exept first one)
+    const [open, setOpen] = useState([true, ...(new Array(versions.length - 1).fill(false))]);
+
+    function handleOnClick(item) {
+        if (!open[item])
+            open[item] = true;
+        else
+            open[item] = false;
```
Comment (JustD4nTe) [inline web-client/src/pages/work-view.js]: Shorter: `open[item] = !open[item] `

File: web-client/src/pages/work-view.js [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+                title = {work.title}
+                path = {work.path}
+                alternativeText = {work.alternativeText}
+                author = {work.author}
+                degree = {work.degree}
+                university = {work.university}
+                authors = {work.authors}
+                text = {work.text}
+            />
+            }
```
Comment (JustD4nTe) [inline web-client/src/pages/work-view.js:309]: Where is the difference?

File: web-client/src/pages/work-view.js [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+                );
+            }
+        }
+        if(user === "reviewer")
+        {
+            reviewsList.push(
+                <CurrentVersion
+                    path={version.reviewer.path}
+                    stars={version.reviewer.stars}
+                    review={version.reviewer.textReview}
```
Comment (JustD4nTe) [inline web-client/src/pages/work-view.js]: Reviewer should see answers from author :p

File: web-client/src/components/WorkForReviewComponent.jsx [hunk 4]
```diff
@@ -0,0 +1,394 @@
...
+                                    value={value}
+                                    onChange={(event, newValue) => {
+                                        if(newValue !== null)
+                                            setValue(newValue);
+                                    }}
+                                    onChangeActive={(event, newHover) => {
+                                        setHover(newHover);
+                                    }}
+                                />
+                                {value !== null && <Box ml={2} className={style.label}>{labels[hover !== -1 ? hover : value]}</Box>}
```
Comment (JustD4nTe) [inline web-client/src/components/WorkForReviewComponent.jsx]: How it works?

File: web-client/src/components/WorkForReviewComponent.jsx [hunk 5]
```diff
@@ -0,0 +1,394 @@
...
+                            }}
+                            multiline
+                            rows={6}
+                            placeholder="Add text"
+                            onChange={handleChange("comment")}
+                            variant="outlined"
+                            error={!!errors.comment}
+                            helperText={errors?.comment?.message}
+                        />
+                        </DialogContent>
```
Comment (JustD4nTe) [inline web-client/src/components/WorkForReviewComponent.jsx]: Document is not formatted corectly => `alt + shift + f`

File: web-client/src/components/WorkForReviewComponent.jsx [hunk 6]
```diff
@@ -0,0 +1,394 @@
...
+                    <form
+                        className={style.form}
+                        noValidate
+                        onSubmit={handleSubmit((data) => {
+                        data.stars = value
+                        alert(JSON.stringify(data))
+                        })}
+                    >
+                        <DialogTitle id='alert-dialog-title' className={style.dialogTitle}>
+                            {'Add your review'}
```
Comment (JustD4nTe) [inline web-client/src/components/WorkForReviewComponent.jsx]: `{'Add your review'}` =>  `Add your review`

File: web-client/src/components/WorkForReviewComponent.jsx [hunk 7]
```diff
@@ -0,0 +1,394 @@
...
+        setOpen(false);
+    };
+
+    const handleChange = (prop) => (event) => {
+        setValues({ ...values, [prop]: event.target.value });
+        setCounts({ ...counts, [prop]: event.target.value.length });
+    };
+
+    //Form
+    const maxCommentSize = 255;
```
Comment (JustD4nTe) [inline web-client/src/components/WorkForReviewComponent.jsx]: Hmn?

File: web-client/src/components/WorkForReviewComponent.jsx [hunk 8]
```diff
@@ -0,0 +1,394 @@
...
+        3: 'accepted'
+    };
+
+    const style = styles();
+
+    const [open, setOpen] = React.useState(false);
+    const [value, setValue] = React.useState(1);
+    const [hover, setHover] = React.useState(1);
+
+    const [_file, SetFile] = React.useState(null);
```
Comment (JustD4nTe) [inline web-client/src/components/WorkForReviewComponent.jsx]: `_file` => `file`

File: web-client/src/components/CurrentVersionWithReplyToReview.jsx [hunk 9]
```diff
@@ -0,0 +1,254 @@
...
+            float: 'left',
+            marginLeft: '2%',
+            marginTop: '3%',
+        },
+    });
+
+    const style = styles();
+    
+    const [value] = useState(props.stars);  
+    const [hover] = useState(3);
```
Comment (JustD4nTe) [inline web-client/src/components/CurrentVersionWithReplyToReview.jsx]: How about this?
```suggestion
    const value = props.stars;  
    const hover = 3;
```

But still I'm little confused with these variables :/ Especially with `values`, do we need it?

File: web-client/src/components/CurrentVersion.jsx [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+        },
+    });
+
+    const style = styles();
+
+    let d = new Date();
+    let hours = `${d.getHours()}:${d.getMinutes()}`;
+    let date = `${d.getDate()}/${d.getMonth()+1}/${d.getFullYear()}`;
+
+    const [hover] = React.useState(1);
```
Comment (JustD4nTe) [inline web-client/src/components/CurrentVersion.jsx]: Same as line 134 in CurrentVersionWithReplyToReview.jsx

**Review by JustD4nTe — CHANGES_REQUESTED** [review]
It looks almost okay for me, because I'm not sure if it'll work properly with our API :/
Anyway, we can merge this (after code formatting) and when we'll have an endpoint in API for this, we'll change it

File: web-client/src/pages/work-view.js (hunk 1 above)
Comment (Kwolik) [reply web-client/src/pages/work-view.js]: ok

File: web-client/src/pages/work-view.js (hunk 2 above)
Comment (Kwolik) [reply web-client/src/pages/work-view.js:309]: WorkForReviewComponent has another button that opens the popup and is created in that component

File: web-client/src/components/WorkForReviewComponent.jsx (hunk 5 above)
Comment (Kwolik) [reply web-client/src/components/WorkForReviewComponent.jsx]: ok sry

File: web-client/src/components/WorkForReviewComponent.jsx (hunk 6 above)
Comment (Kwolik) [reply web-client/src/components/WorkForReviewComponent.jsx]: ok

File: web-client/src/components/WorkForReviewComponent.jsx (hunk 7 above)
Comment (Kwolik) [reply web-client/src/components/WorkForReviewComponent.jsx]: ?

File: web-client/src/components/WorkForReviewComponent.jsx (hunk 4 above)
Comment (Kwolik) [reply web-client/src/components/WorkForReviewComponent.jsx]: image

displays the appropriate star value [rejected, correct, accepted]

File: web-client/src/components/WorkForReviewComponent.jsx (hunk 7 above)
Comment (Kwolik) [reply web-client/src/components/WorkForReviewComponent.jsx]: information for me when the elements related to form validation begin

File: web-client/src/components/CurrentVersionWithReplyToReview.jsx (hunk 9 above)
Comment (Kwolik) [reply web-client/src/components/CurrentVersionWithReplyToReview.jsx]: yes you are right i even changed to const hover = props.stars;

You're right so I deleted it

File: web-client/src/components/CurrentVersion.jsx (hunk 10 above)
Comment (Kwolik) [reply web-client/src/components/CurrentVersion.jsx]: You're right so I deleted it

File: web-client/src/pages/work-view.js (hunk 3 above)
Comment (Kwolik) [reply web-client/src/pages/work-view.js]: Done 👍

**Review by upsu90 — APPROVED** [review]

**Review by JustD4nTe — APPROVED** [review]

File: web-client/src/components/WorkForReviewComponent.jsx [hunk 11]
```diff
@@ -0,0 +1,391 @@
...
+    title: 0,
+    description: 0,
+  });
+
+  const schema = yup.object().shape({
+    file: yup.string(),
+    comment: yup
+      .string()
+      .matches(
+        /^[A-Za-z0-9]*$/,
```
Comment (KamilDonda) [inline web-client/src/components/WorkForReviewComponent.jsx]: The comment should also contain spaces and characters like ```: ,   . -  ? ! ( ) ' "```
image

File: web-client/src/components/CurrentVersionWithReplyToReview.jsx [hunk 12]
```diff
@@ -0,0 +1,251 @@
...
+    const handleChange = (prop) => (event) => {
+        setValues({ ...values, [prop]: event.target.value });
+        setCounts({ ...counts, [prop]: event.target.value.length });
+    };
+
+    const schema = yup.object().shape({
+        comment: yup
+        .string()
+        .matches(
+            /^[A-Za-z0-9]*$/,
```
Comment (KamilDonda) [inline web-client/src/components/CurrentVersionWithReplyToReview.jsx]: Same as above
image

File: web-client/src/components/CurrentVersionWithReplyToReview.jsx [hunk 13]
```diff
@@ -0,0 +1,251 @@
...
+                                }}
+                                multiline
+                                rows={6}
+                                placeholder="Add comment"
+                                onChange={handleChange("comment")}
+                                variant="outlined"
+                                error={!!errors.comment}
+                                helperText={errors?.comment?.message}
+                            />
+                            <span className={style.info}>Remember: you can't edit or delete this commen</span>
```
Comment (KamilDonda) [inline web-client/src/components/CurrentVersionWithReplyToReview.jsx]: commen -> comment

**Review by KamilDonda — APPROVED** [review]

**Pull request merged**